### DISCUSSION TITLE
remix project template for deno

### DISCUSSION THREAD

**Discussion by elycheikhsmail** [remix-run/remix#2207] [Proposals]
I hope to get template for start new remix project using deno and deno ecosystem instead of node. I think now is possible to write  js code and run it with both js runtime deno and node ( dnt is js module that convert js writen for deno to js code compatible ).
this will improve learning js frontend and add feature to remix core more easier in long term

**Comment by lensbart** [comment]
Is https://github.com/jacob-ebey/remix-deno what you’re looking for?

@jacob-ebey out of curiosity, is this repository the future direction for `create-remix` when using Deno Deploy as deployment target? Thanks!

**Reply by john-griffin** [reply]
This example supports running remix via Deno's JIT https://github.com/jacob-ebey/remix-deno-jit

I was even able to deploy it to Deno Deploy.

**Comment by MichaelDeBoey** [comment]
We have https://github.com/remix-run/remix/tree/main/templates/deno

You can install it by running `npx create-remix@latest --template deno`

**Reply by john-griffin** [reply]
Yeah that was what I tried at first but it still requires Node for tooling (it comes with a package.json). Using https://github.com/jacob-ebey/remix-deno-jit seems closer to the ideal Deno dev experience. 

Is there any chance that the Deno JIT approach would be considered for upstreaming an official Remix template?

**Reply by MichaelDeBoey** [reply]
We're open for PRs, so if you think you can update the `deno` template, please create a PR to update it

**Reply by john-griffin** [reply]
I just noticed the decision doc https://github.com/remix-run/remix/blob/main/decisions/0001-use-npm-to-manage-npm-dependencies-for-deno-projects.md#vs-code-type-hints

Seems like the decision has already been made to just focus on node/npm tooling for Deno projects?

**Reply by MichaelDeBoey** [reply]
Oh yeah forgot about that one 🙈

If you want to have a non-npm Deno template, you can of course always create your own custom stack
https://remix.run/stacks#custom-stacks

**Reply by pcattori** [reply]
One thing to note is that we wrote that decision doc before Deno supported imports from `npm`. Haven't had time to revisit that to see if the peer dependency issues we were running into have been address in Deno or not.

**Reply by john-griffin** [reply]
Peer dependencies are now supported in Deno after this PR https://github.com/denoland/deno/pull/16561

This was shipped in Deno [v1.27.2](https://github.com/denoland/deno/releases/tag/v1.27.2)

Is that enough progress to reevaluate this decision?

**Reply by pcattori** [reply]
Yes, I think so. We still have to see how treeshaking and dev/prod/test environments are handled in Deno. Still need to check if NPM support in Deno addresses the treeshaking and dev/prod/test env concerns we detailed in the decision doc. But yea I think its a good time to recheck those.

Probably won't have time to look into this in the next few weeks, so if someone wants to look into it before then, that'd be more than welcome.

Specifically, we'd want a Deno-native DX for users writing their Remix apps with Deno. So ideally no need to install `node`/`npm`, no `node_modules/`, etc..

**Reply by IshanKBG** [reply]
Any update from the team?

**Reply by algoflows** [reply]
Yo what happened here, any remix docs coverage for deploying with Deno.

**Reply by TM7-MStar** [reply]
https://github.com/remix-run/remix/tree/main/templates/deno is dead and https://github.com/redabacha/remix-deno-vite-template uses node_modules, probably because Vite requires node. Was hoping to find a Remix template which eliminates the need for node altogether and runs on Deno natively without adapters.

**Reply by nikolailehbrink** [reply]
According to their [1.x to 2.x Migration Guide](https://docs.deno.com/runtime/reference/migration_guide/#framework-support), Remix is now supported, because of the improved Node.js and npm compatibility.
I just tried to scaffold out a new project with `deno -A npm:create-remix@latest` as shown in their [Announcement video](https://www.youtube.com/live/d35SlRgVxT8?si=u6Ie4Ix70Vd9iwlT&t=851) and it works. 

I know it's not the native solution @TM7-MStar for example is looking for, but just wanted to leave it here, if anyone is interested.

**Reply by TimSusa** [reply]
I cannot see your point as by
```
 deno -A npm:create-remix@latest 
```
 it provides me with a good old npm repo for remix without any deno.json or deno specific. Maybe I did not get it, but it is useless in that concern. However, if I try 
```
deno run -A npm:create-remix@latest --no-install --template redabacha/remix-deno-vite-template
```
at least I get provided a runnable deno.json. Here my question, why that guys still have that package.json project in it? Is that state of the art, or what?

**Comment by nikolailehbrink** [comment]
An official `deno` template can now be found for React Router 7 here: https://github.com/remix-run/react-router-templates/tree/main/deno

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
